### PR TITLE
Some refactoring of CloudBigtableIO objects.

### PR DESCRIPTION
**Pull request by sduskis**
Moving the Reader class out of the Source class, so there's less nesting.
Using superclasses in tests and CloudBigtableIO to reduce leaking internals.
Adding createSourceWithKeys() instead of using new.

All of these changes are steps to getting to #672

### CODE REVIEW

File: bigtable-hbase-dataflow/src/test/java/com/google/cloud/bigtable/dataflow/CloudBigtableIOIntegrationTest.java [hunk 1]
```diff
@@ -224,7 +225,7 @@ public void testEstimatedAndSplitForSmallTable() throws Exception {
         CloudBigtableScanConfiguration config =
             new CloudBigtableScanConfiguration(projectId, zoneId, clusterId,
                 tableName.getQualifierAsString(), new Scan());
-        CloudBigtableIO.Source source = new CloudBigtableIO.Source(config);
+        CloudBigtableIO.Source source = (Source) CloudBigtableIO.read(config);
```
Comment (mgarolera) [inline bigtable-hbase-dataflow/src/test/java/com/google/cloud/bigtable/dataflow/CloudBigtableIOIntegrationTest.java]: Why the change from new Source to .read and cast?

Comment (sduskis) [inline bigtable-hbase-dataflow/src/test/java/com/google/cloud/bigtable/dataflow/CloudBigtableIOIntegrationTest.java]: This does indeed sound like a code smell.  The idea of CloudBigtableIO.read() should just be to get the generic Dataflow object, but we're reaching under the covers here.

We're actually trying to test the sample row key processing, and need to reach under the covers to get to it.  I should really move the logic and this test to separate classes.  I'll add a TODO for this PR.

File: bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableIO.java [hunk 2]
```diff
@@ -566,13 +508,78 @@ public boolean producesSortedKeys(PipelineOptions options) throws Exception {
...
+        List<BoundedSource<Result>> result = new ArrayList<>();
         for (int i = 0; i < splitCount; i++) {
-          result.add(new SourceWithKeys(splitKeys[i], splitKeys[i + 1], regionSize));
+          result.add(createSourceWithKeys(splitKeys[i], splitKeys[i + 1], regionSize));
         }
         return result;
       }
     }
+
+    public BoundedSource<Result> createSourceWithKeys(byte[] startKey, byte[] stopKey, long size) {
```
Comment (kevinsi4508) [inline bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableIO.java]: package default, for testing only?

Comment (kevinsi4508) [inline bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableIO.java]: if not, can it be public static?

**Comment by kevinsi4508** [comment]
LGTM
